packages/test/test-service-load/src/leaderElection.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IFluidDataStoreRuntime } from "@fluidframework/datastore-definitions";
+import { ISignalMessage } from "@fluidframework/protocol-definitions";
+
+export class LeaderElection {
+    private readonly beatInEveryNSecs: number = 1000; // 1 secs
+    private readonly leaderWait: number = 5000; // 5 secs
+    private lastPinged: Date | undefined;
+
+    constructor(private readonly dataStoreRuntime: IFluidDataStoreRuntime) { }
+
+    public setupLeaderElection() {
+        this.dataStoreRuntime.on("signal", (signal: ISignalMessage) => this.handleSignal(signal));
+
+        const interval = setInterval(() => {
+            if (this.leaderId !== undefined && this.leaderId === this.dataStoreRuntime.clientId) {
+                this.dataStoreRuntime.submitSignal("leaderMessage", "leaderMessage");
+                this.lastPinged = new Date();
+            }else if(this.leaderId === undefined) {
+                this.dataStoreRuntime.logger.send(
```

Comment:
i think this is super close. rather than directly using the runtime logger, can we create our own child logger over it? ChildLogger.create. this can then have something like SignalLeaderElection as the namespace. it will also give you the send error method which will properly format your error